input/system/selfhosted/otel_handler.go

```diff
@@ -0,0 +1,211 @@
+		}
+		if rv.Key == "labels" {
+			for _, ll := range rv.Value.GetKvlistValue().Values {
+				k8sLabels[ll.Key] = ll.Value.GetStringValue()
+			}
+		}
+	}
+
+	if server.Config.LogOtelK8SPod != "" {
+		parts := strings.SplitN(server.Config.LogOtelK8SPod, "/", 2)
+		if len(parts) == 2 {
+			if k8sNamespaceName != parts[0] || k8sPodName != parts[1] {
+				return true
+			}
+		} else if len(parts) == 1 {
+			if k8sPodName != parts[0] {
+				return true
+			}
+		} else {
+			prefixedLogger.PrintWarning("Pod specification for OTel server not valid (need zero or one / separator): \"%s\", skipping log record\n", server.Config.LogOtelK8SPod)
+			return true
+		}
+	}
+
+	if server.Config.LogOtelK8SLabels != "" {
+		selectors := strings.Split(server.Config.LogOtelK8SLabels, ",")
+		for _, selector := range selectors {
+			parts := k8sSelectorRegexp.FindStringSubmatch(selector)
+			if parts != nil {
+				selKey := parts[1]
+				selEq := parts[2] == "=" || parts[2] == "=="
+				selNotEq := parts[2] == "!="
+				selValue := parts[3]
+				v, ok := k8sLabels[selKey]
+				if ok {
+					if (selEq && v != selValue) || (selNotEq && v == selValue) {
+						return true
+					}
+				}
+			} else {
+				prefixedLogger.PrintWarning("Label selector for OTel server not valid: \"%s\", skipping log record\n", server.Config.LogOtelK8SLabels)
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func setupOtelHandler(ctx context.Context, server *state.Server, rawLogStream chan<- SelfHostedLogStreamItem, parsedLogStream chan state.ParsedLogStreamItem, prefixedLogger *util.Logger) error {
+	otelLogServer := server.Config.LogOtelServer
+	tz := server.GetLogTimezone()
+	go func() {
+		http.HandleFunc("/v1/logs", func(w http.ResponseWriter, r *http.Request) {
+			b, err := io.ReadAll(r.Body)
+			if err != nil {
+				prefixedLogger.PrintError("Could not read otel body")
+			}
+			logsData := &otlpLogs.LogsData{}
+			if err := proto.Unmarshal(b, logsData); err != nil {
+				prefixedLogger.PrintError("Could not unmarshal otel body")
+			}
+			for _, r := range logsData.ResourceLogs {
+				for _, s := range r.ScopeLogs {
+					for _, l := range s.LogRecords {
+						var logger string
+						var record *common.KeyValueList
+						var kubernetes *common.KeyValueList
+						hasErrorSeverity := false
+						if l.Body.GetKvlistValue() != nil {
+							for _, v := range l.Body.GetKvlistValue().Values {
+								if v.Key == "logger" {
+									logger = v.Value.GetStringValue()
+								}
+								if v.Key == "record" {
+									record = v.Value.GetKvlistValue()
+								}
+								if v.Key == "kubernetes" {
+									kubernetes = v.Value.GetKvlistValue()
+								}
+								if v.Key == "error_severity" {
+									hasErrorSeverity = true
+								}
+							}
+							// TODO: Does the logger name need to be configurable? (this is tested with CNPG,
+							// but would the logger be different with other operators/container names?)
+							if logger == "postgres" {
```

Comment:
I tweaked here to something like the following and I was able to make it work with Crunchy + `logging_collector` turned off:

```
							if handleWithK8sFilter(kubernetes, server) {
								prefixedLogger.PrintInfo("Received logs matching to the filter\n")
								// handle log as plain log for now
								// "log"=>"2024-03-22 09:40:37.089 UTC [320] [user=postgres,db=postgres,app=psql] LOG:  duration: 13013.199 ms  statement: select pg_sleep(13);"
								item := SelfHostedLogStreamItem{}
								item.Line = log
								item.OccurredAt = time.Unix(0, int64(l.TimeUnixNano))
								rawLogStream <- item
							} else if logger == "postgres" {
```

That said, I don't think this is something people would do (though I think it's still good to introduce the "match filter" that's what I did above with `handleWithK8sFilter`, as opposed to "deny filter" that this PR currently has).
I'd love to learn more how people are doing logs with Crunchy postgres operator. There is a blog post about it using vector (https://www.crunchydata.com/blog/log-export-examples-using-crunchy-postgres-for-kubernetes), which we can try but I think vector doesn't have otel sink.